File: game_server/ServerProtocol/src/GS2GC/p002_InitOp/GS2GC_002_015_RetConsortList.java
package GS2GC.p002_InitOp;

import java.nio.ByteBuffer;
/*********
 * 家人初始化
 **/
public class GS2GC_002_015_RetConsortList implements ALBasicProtocolPack._IALProtocolStructure {
/** 已获得家人列表 */
private java.util.ArrayList<Common.ConsortObj.Consort_Info> consortList;
/** 已解锁的CG列表 */
private java.util.ArrayList<Common.ConsortObj.Consort_CGInfo> unlockedCGList;
/** 随机邀约的指定妃子ID列表 */
private java.util.ArrayList<Long> randCallConsortIdList;


public GS2GC_002_015_RetConsortList() {
	consortList = new java.util.ArrayList<Common.ConsortObj.Consort_Info>();
	unlockedCGList = new java.util.ArrayList<Common.ConsortObj.Consort_CGInfo>();
	randCallConsortIdList = new java.util.ArrayList<Long>();
}

public GS2GC_002_015_RetConsortList(
	 java.util.ArrayList<Common.ConsortObj.Consort_Info> _consortList
	, java.util.ArrayList<Common.ConsortObj.Consort_CGInfo> _unlockedCGList
	, java.util.ArrayList<Long> _randCallConsortIdList
) {	consortList = _consortList;
	unlockedCGList = _unlockedCGList;
	randCallConsortIdList = _randCallConsortIdList;
}

public final byte getMainOrder() { return (byte)2; }

public final byte getSubOrder() { return (byte)15; }

/** 已获得家人列表 */
public java.util.ArrayList<Common.ConsortObj.Consort_Info> getConsortList() { return consortList; }
/** 已获得家人列表 */
public void addConsortList(Common.ConsortObj.Consort_Info _consortList) { consortList.add(_consortList); }
/** 已解锁的CG列表 */
public java.util.ArrayList<Common.ConsortObj.Consort_CGInfo> getUnlockedCGList() { return unlockedCGList; }
/** 已解锁的CG列表 */
public void addUnlockedCGList(Common.ConsortObj.Consort_CGInfo _unlockedCGList) { unlockedCGList.add(_unlockedCGList); }
/** 随机邀约的指定妃子ID列表 */
public java.util.ArrayList<Long> getRandCallConsortIdList() { return randCallConsortIdList; }
/** 随机邀约的指定妃子ID列表 */
public void addRandCallConsortIdList(long _randCallConsortIdList) { randCallConsortIdList.add(_randCallConsortIdList); }


public final int GetBufSize() {
	int _size = 0;
	_size += 2;
	for(int _i = 0; _i < consortList.size(); _i++) {
	_size += 4 + consortList.get(_i).GetBufSize();
	}

	_size += 2 + (unlockedCGList.size() * 13);
	_size += 2 + (randCallConsortIdList.size() * 8);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 2;
	_size += 2;
	for(int _i = 0; _i < consortList.size(); _i++) {
	_size += 4 + consortList.get(_i).GetBufSize();
	}

	_size += 2 + (unlockedCGList.size() * 13);
	_size += 2 + (randCallConsortIdList.size() * 8);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _consortListCount = _buf.getShort();
	for(int _i = 0; _i < _consortListCount; _i++) { 
		Common.ConsortObj.Consort_Info _consortList = new Common.ConsortObj.Consort_Info();
		if(_buf.remaining() <= 0) return;
	int __consortListCustLen = _buf.getInt();
	int __consortListCurPos = _buf.position();
	_consortList.ReadUnzipBuf(_buf, __consortListCurPos + __consortListCustLen);
	_buf.position(__consortListCurPos + __consortListCustLen);

		consortList.add(_consortList);
	}
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _unlockedCGListCount = _buf.getShort();
	for(int _i = 0; _i < _unlockedCGListCount; _i++) { 
		Common.ConsortObj.Consort_CGInfo _unlockedCGList = new Common.ConsortObj.Consort_CGInfo();
		if(_buf.remaining() <= 0) return;
	int __unlockedCGListCustLen = _buf.getInt();
	int __unlockedCGListCurPos = _buf.position();
	_unlockedCGList.ReadUnzipBuf(_buf, __unlockedCGListCurPos + __unlockedCGListCustLen);
	_buf.position(__unlockedCGListCurPos + __unlockedCGListCustLen);

		unlockedCGList.add(_unlockedCGList);
	}
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _randCallConsortIdListCount = _buf.getShort();
	for(int _i = 0; _i < _randCallConsortIdListCount; _i++) { 
		long _randCallConsortIdList = (long)0;
		if(_buf.remaining() > 0) _randCallConsortIdList = _buf.getLong();
		randCallConsortIdList.add(_randCallConsortIdList);
	}
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putShort((short)consortList.size());
	for(int _i = 0; _i < consortList.size(); _i++) { 
		_buf.putInt(consortList.get(_i).GetBufSize());
	consortList.get(_i).PutUnzipBuf(_buf);
	}
	_buf.putShort((short)unlockedCGList.size());
	for(int _i = 0; _i < unlockedCGList.size(); _i++) { 
		_buf.putInt(unlockedCGList.get(_i).GetBufSize());
	unlockedCGList.get(_i).PutUnzipBuf(_buf);
	}
	_buf.putShort((short)randCallConsortIdList.size());
	for(int _i = 0; _i < randCallConsortIdList.size(); _i++) { 
		_buf.putLong(randCallConsortIdList.get(_i));
	}
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)2);
	_buf.put((byte)15);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)2);
	_recBuf.put((byte)15);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

File: game_server/ClientProtocol/GS2GC/p002_InitOp/GS2GC_002_015_RetConsortList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GS2GC.p002_InitOp
{

/// <summary>
/// 家人初始化
/// </summary>
public class GS2GC_002_015_RetConsortList : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 已获得家人列表
/// </summary>
private List<Common.ConsortObj.Consort_Info> consortList;
/// <summary>
/// 已解锁的CG列表
/// </summary>
private List<Common.ConsortObj.Consort_CGInfo> unlockedCGList;
/// <summary>
/// 随机邀约的指定妃子ID列表
/// </summary>
private List<long> randCallConsortIdList;


public GS2GC_002_015_RetConsortList() {
	consortList = new List<Common.ConsortObj.Consort_Info>();
	unlockedCGList = new List<Common.ConsortObj.Consort_CGInfo>();
	randCallConsortIdList = new List<long>();
}

public GS2GC_002_015_RetConsortList(
	List<Common.ConsortObj.Consort_Info> _consortList
	, List<Common.ConsortObj.Consort_CGInfo> _unlockedCGList
	, List<long> _randCallConsortIdList
) {	consortList = _consortList;
	unlockedCGList = _unlockedCGList;
	randCallConsortIdList = _randCallConsortIdList;
}

public byte getMainOrder() { return (byte)2; }

public byte getSubOrder() { return (byte)15; }

/// <summary>
/// 已获得家人列表
/// </summary>
public List<Common.ConsortObj.Consort_Info> getConsortList() { return consortList; }
/// <summary>
/// 已获得家人列表
/// </summary>
public void addConsortList(Common.ConsortObj.Consort_Info _consortList) { consortList.Add(_consortList); }
/// <summary>
/// 已解锁的CG列表
/// </summary>
public List<Common.ConsortObj.Consort_CGInfo> getUnlockedCGList() { return unlockedCGList; }
/// <summary>
/// 已解锁的CG列表
/// </summary>
public void addUnlockedCGList(Common.ConsortObj.Consort_CGInfo _unlockedCGList) { unlockedCGList.Add(_unlockedCGList); }
/// <summary>
/// 随机邀约的指定妃子ID列表
/// </summary>
public List<long> getRandCallConsortIdList() { return randCallConsortIdList; }
/// <summary>
/// 随机邀约的指定妃子ID列表
/// </summary>
public void addRandCallConsortIdList(long _randCallConsortIdList) { randCallConsortIdList.Add(_randCallConsortIdList); }


public int GetBufSize() {
	int _size = 0;
	_size += 2;
for(int _i = 0; _i < consortList.Count; _i++) {
	_size += 4 + consortList[_i].GetBufSize();
	}

	_size += 2 + (unlockedCGList.Count * 13);
	_size += 2 + (randCallConsortIdList.Count * 8);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 2;
	_size += 2;
for(int _i = 0; _i < consortList.Count; _i++) {
	_size += 4 + consortList[_i].GetBufSize();
	}

	_size += 2 + (unlockedCGList.Count * 13);
	_size += 2 + (randCallConsortIdList.Count * 8);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _consortListCount = _buf.getShort();
	for(int _i = 0; _i < _consortListCount; _i++) { 
		Common.ConsortObj.Consort_Info _consortList = new Common.ConsortObj.Consort_Info();
		int __consortListCustLen = _buf.getInt();
	int __consortListCurPos = _buf.getCurPos();
	_consortList.ReadUnzipBuf(_buf, __consortListCurPos + __consortListCustLen);
	_buf.setPosition(__consortListCurPos + __consortListCustLen);

		consortList.Add(_consortList);
	}
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _unlockedCGListCount = _buf.getShort();
	for(int _i = 0; _i < _unlockedCGListCount; _i++) { 
		Common.ConsortObj.Consort_CGInfo _unlockedCGList = new Common.ConsortObj.Consort_CGInfo();
		int __unlockedCGListCustLen = _buf.getInt();
	int __unlockedCGListCurPos = _buf.getCurPos();
	_unlockedCGList.ReadUnzipBuf(_buf, __unlockedCGListCurPos + __unlockedCGListCustLen);
	_buf.setPosition(__unlockedCGListCurPos + __unlockedCGListCustLen);

		unlockedCGList.Add(_unlockedCGList);
	}
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _randCallConsortIdListCount = _buf.getShort();
	for(int _i = 0; _i < _randCallConsortIdListCount; _i++) { 
		long _randCallConsortIdList = (long)0;
		_randCallConsortIdList = _buf.getLong();
		randCallConsortIdList.Add(_randCallConsortIdList);
	}
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putShort((short)consortList.Count);
	for(int _i = 0; _i < consortList.Count; _i++) { 
		_buf.putInt(consortList[_i].GetBufSize());
	consortList[_i].PutUnzipBuf(_buf);
	}
	_buf.putShort((short)unlockedCGList.Count);
	for(int _i = 0; _i < unlockedCGList.Count; _i++) { 
		_buf.putInt(unlockedCGList[_i].GetBufSize());
	unlockedCGList[_i].PutUnzipBuf(_buf);
	}
	_buf.putShort((short)randCallConsortIdList.Count);
	for(int _i = 0; _i < randCallConsortIdList.Count; _i++) { 
		_buf.putLong(randCallConsortIdList[_i]);
	}
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)2);
	_buf.put((byte)15);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)2);
	_recBuf.put((byte)15);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("consortList").Append(":").Append(consortList.ToString()).Append(", ");
	builder.Append("unlockedCGList").Append(":").Append(unlockedCGList.ToString()).Append(", ");
	builder.Append("randCallConsortIdList").Append(":").Append(randCallConsortIdList.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

